File: 53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n=len(nums)
        maxs=float('-inf')
        sum=0
        for i in nums:
            sum+=i
            maxs=max(maxs,sum)
            if sum<0:
                sum=0
        return maxs 
        

File: DSA_Java/53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=0;
        int maxs=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            maxs=Math.max(sum,maxs);
            if (sum < 0){
                sum=0;
            }
        }
        return maxs;
        
    }
}